plugins/@grouparoo/app-templates/src/destination/shared/batch.ts

```diff
@@ -0,0 +1,93 @@
+import { ErrorWithProfileGuid } from "@grouparoo/core";
+
+export interface MethodNormalizeGroupName {
+  // mess with the names of groups (tags with no spaces, for example)
+  (argument: { groupName: string; config: any }): string;
+}
+
+export interface GroupNameExport {
+  oldGroups?: string[];
+  newGroups?: string[];
+  addedGroups?: string[];
+  removedGroups?: string[];
+  toDelete?: boolean;
+}
+
+export function setGroupNames(
+  exportedProfile: GroupNameExport,
+  methods: {
+    normalizeGroupName?: MethodNormalizeGroupName;
+  },
+  config: any
+) {
+  // build up groups situation of group names to addition and removal
+  let oldGroups = exportedProfile.oldGroups || [];
+  let newGroups = exportedProfile.newGroups || [];
+  if (exportedProfile.toDelete) {
+    // if anyone asks, removing from all groups as part of deletion
+    oldGroups = Array.from(new Set(oldGroups.concat(newGroups)));
+    newGroups = [];
```

Comment:
Think this should be something that core does when `toDelete` is true?

Reply:
Not sure. For now, I lean on the side of leaving it as it is. knowing which groups they were actually sent (oldGroups) before.